src/mscorlib/src/System/IO/StreamReader.cs

```diff
@@ -529,6 +532,22 @@ private bool IsPreamble()
             return _checkPreamble;
         }
 
+        private StringBuilder AcquireStringBuilder(int capacity)
+        {
+            var sb = this.builder;
+
+            if (sb == null)
+                this.builder = sb = new StringBuilder(capacity);
+            else
+            {
+                sb.Length = 0;
```

Comment:
It would be good to add a few comments here around what Length=0 does (essentially clears the string builder) and what the capacity check does (increasing the capacity re-allocates the backing storage, so we only re-allocate when the request is larger than all previous requests).